code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm

```diff
@@ -0,0 +1,621 @@
+	action_icon_state = "warp_shadow"
+	mechanics_text = "Binds our psychic essence to a spot of our choosing. We can use Hyperposition to swap locations with this essence."
+	plasma_cost = 150
+	cooldown_timer = 60 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_PLACE_WARP_BEACON
+	/// This is the warp shadow that the Wraith creates with its Place Warp Shadow ability, and teleports to with Hyperposition
+	var/obj/effect/xenomorph/warp_shadow/warp_shadow
+
+/datum/action/xeno_action/place_warp_shadow/can_use_action(silent = FALSE, override_flags)
+	. = ..()
+	if(!check_warp_shadow_placement(get_turf(owner), owner)) //Check if our placement is legal
+		if(!silent)
+			to_chat(owner, "<span class='xenowarning'>We cannot create a warp shadow here!</span>")
+		return FALSE
+
+/datum/action/xeno_action/place_warp_shadow/action_activate()
+	var/mob/living/carbon/xenomorph/wraith/ghost = owner
+
+	ghost.visible_message("<span class='xenowarning'>The air starts to violently roil and shimmer around [ghost]!</span>", \
+	"<span class='xenodanger'>We begin to imprint our essence upon reality, causing the air about us to roil and shimmer...</span>") //Fluff
+
+	if(!do_after(ghost, WRAITH_PLACE_WARP_BEACON_WINDUP, TRUE, ghost, BUSY_ICON_BUILD)) //Channel time/wind up
+		ghost.visible_message("<span class='xenowarning'>The space around [ghost] abruptly stops shifting and wavering.</span>", \
+		"<span class='xenodanger'>We cease binding our essence to this place...</span>")
+		add_cooldown(cooldown_override = WRAITH_PLACE_WARP_BEACON_FAIL_COOLDOWN_OVERRIDE)
+		return fail_activate()
+
+	var/turf/T = get_turf(ghost)
+	ghost.visible_message("<span class='xenonotice'>\A shimmering point suddenly coalesces from the warped space above [T].</span>", \
+	"<span class='xenodanger'>We complete our work, binding our essence to this point.</span>", null, 5) //Fluff
+	var/obj/effect/xenomorph/warp_shadow/shadow = new(T) //Create the new warp shadow.
+	playsound(T, 'sound/weapons/emitter.ogg', 25, 1)
+	QDEL_NULL(warp_shadow) //Delete the old warp shadow
+	warp_shadow = shadow //Set the new warp shadow
+	RegisterSignal(owner, COMSIG_XENOMORPH_DEATH, .proc/unset_warp_shadow) //Removes warp shadow on death
+	RegisterSignal(warp_shadow, COMSIG_PARENT_PREQDELETED, .proc/unset_warp_shadow) //For var clean up
+	RegisterSignal(owner, COMSIG_PARENT_PREQDELETED, .proc/unset_warp_shadow) //For var clean up
+	warp_shadow.setDir(ghost.dir) //Have it imitate our facing
+	warp_shadow.pixel_x = ghost.pixel_x //Inherit pixel offsets
+	warp_shadow.pixel_y = ghost.pixel_y //Inherit pixel offsets
+	teleport_debuff_aoe(warp_shadow) //SFX
+
+	succeed_activate()
+	add_cooldown()
+
+///Nulls out the warp shadow
+/datum/action/xeno_action/place_warp_shadow/proc/unset_warp_shadow()
+	SIGNAL_HANDLER
+	UnregisterSignal(warp_shadow, COMSIG_PARENT_PREQDELETED)
+	UnregisterSignal(owner, COMSIG_PARENT_PREQDELETED)
+	QDEL_NULL(warp_shadow) //remove the actual shadow
+	warp_shadow = null
```

Comment:
unneeded

Reply:
I do need to get rid of the warp shadow at least